cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -1170,8 +1170,7 @@ private Future<MapAndStatus<ConfigMap, KafkaRebalanceStatus>> onReady(Reconcilia
                             || kafka.getStatus().getConditions() == null
                             || kafka.getStatus().getConditions().stream().noneMatch(condition -> condition.getType().equals("Ready") && condition.getStatus().equals("True"))) {
```

Comment:
@kanha-gupta You are now updating the status as `null` which means that even if `KafkaCluster` was `NotReady` from the start, it will not reflecting any updates in the `KafkaRebalance` CR status. As per the issue we just want to stop the transition of `KafkaRebalance` from `Ready` to `NotReady` state  if the kafka resource becomes `NotReady` and the kafka rebalance resource was already in `Ready` state. So ideally you can place some check in the `if` condition above(line 1169-1171) for checking if the rebalance resource is ready or not.